ipinfo/ipinfo.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipinfo
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+type IPInfoMap interface {
+	GetIPInfo(net.IP) (IPInfo, error)
+}
+
+type IPInfo struct {
+	CountryCode CountryCode
+	ASN         int
+}
+
+type CountryCode string
+
+func (cc CountryCode) String() string {
+	return string(cc)
+}
+
+const (
+	errParseAddr     CountryCode = "XA"
```

Comment:
`XA`, `XL` and `XD` seem to be valid country codes: https://docs.dnb.com/partner/en-US/iso_country_codes. Why are we using these codes to represent errors?